server/mdm/mdm.go

```diff
@@ -23,3 +24,24 @@ func DecryptBase64CMS(p7Base64 string, cert *x509.Certificate, key crypto.Privat
 
 	return p7.Decrypt(cert, key)
 }
+
+func GetRawProfilePlatform(profile []byte) string {
+	// trim leading whitespaces
+	trimmedProfile := bytes.TrimSpace(profile)
+
+	if len(trimmedProfile) == 0 {
+		return ""
+	}
+
+	darwinPrefix := []byte("<?xml")
+	if bytes.EqualFold(darwinPrefix, trimmedProfile[:len(darwinPrefix)]) {
```

Comment:
Oh sorry if that was not clear in my previous comment, you probably still need the length check otherwise the `trimmedProfile[:len(...Prefix)]` could panic if it's shorter.